src/App.tsx

```diff
@@ -14,11 +15,104 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  goodListVisible: boolean,
+  filteredGoods: string[]
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    goodListVisible: false,
+    filteredGoods: [...goodsFromServer],
+  };
```

Comment:
leave in state goods or filteredGoods, not two of them
you can use goodsFromServer to reset state's good to initial value